app/assets/javascripts/admin/addon/mixins/setting-component.js

```diff
@@ -103,51 +97,156 @@ export default Mixin.create({
+      "default_email_mailing_list_mode",
+      "default_email_mailing_list_mode_frequency",
+      "default_email_previous_replies",
+      "default_email_in_reply_to",
+      "default_other_new_topic_duration_minutes",
+      "default_other_auto_track_topics_after_msecs",
+      "default_other_notification_level_when_replying",
+      "default_other_external_links_in_new_tab",
+      "default_other_enable_quoting",
+      "default_other_enable_defer",
+      "default_other_dynamic_favicon",
+      "default_other_like_notification_frequency",
+      "default_other_skip_new_user_tips",
+      "default_topics_automatic_unpin",
+      "default_categories_watching",
+      "default_categories_tracking",
+      "default_categories_muted",
+      "default_categories_watching_first_post",
+      "default_categories_regular",
+      "default_tags_watching",
+      "default_tags_tracking",
+      "default_tags_muted",
+      "default_tags_watching_first_post",
+      "default_text_size",
+      "default_title_count_mode",
+    ];
+    const key = this.buffered.get("setting");
+
+    if (defaultUserPreferences.includes(key)) {
+      const data = {};
+      data[key] = this.buffered.get("value");
+
+      ajax(`/admin/site_settings/${key}/user_count.json`, {
+        type: "PUT",
+        data,
+      }).then((result) => {
+        const count = result.user_count;
+
+        if (count > 0) {
+          const controller = showModal("site-setting-default-categories", {
+            model: { count, key: key.replace(/_/g, " ") },
+            admin: true,
+          });
+
+          controller.set("onClose", () => {
+            this.updateExistingUsers = controller.updateExistingUsers;
+            this.send("save");
+          });
+        } else {
           this.send("save");
         }
-      }
+      });
+    } else {
+      this.send("save");
+    }
+  },
+
+  @action
+  save() {
+    this._save()
+      .then(() => {
+        this.set("validationMessage", null);
+        this.commitBuffer();
+        if (AUTO_REFRESH_ON_SAVE.includes(this.setting.setting)) {
+          this.afterSave();
+        }
+      })
+      .catch((e) => {
+        if (e.jqXHR?.responseJSON?.errors) {
+          this.set("validationMessage", e.jqXHR.responseJSON.errors[0]);
+        } else {
+          this.set("validationMessage", I18n.t("generic_error"));
+        }
+      });
+  },
+
+  @action
+  cancel() {
+    this.rollbackBuffer();
+  },
+
+  @action
+  resetDefault() {
+    this.set("buffered.value", this.get("setting.default"));
+  },
+
+  @action
+  toggleSecret() {
+    this.toggleProperty("isSecret");
+  },
+
+  @action
+  setDefaultValues() {
+    this.set(
+      "buffered.value",
+      this.bufferedValues.concat(this.defaultValues).uniq().join("|")
     );
+    return false;
+  },
+
+  @bind
+  _handleKeydown(event) {
+    if (
+      event.key === "Enter" &&
+      event.target.classList.contains("input-setting-string")
+    ) {
+      this.send("save");
```

Comment:
I believe this can now be simply:

```suggestion
      this.save();
```

Reply:
oh yes indeed 👌  missed this, there are others